### PR TITLE
Add unique labels and support`lt` arithmetic command

### PR DESCRIPTION
**Pull request by wildmaples**
### Changes

* Support `lt` arithmetic command
* Add unique labels for both `eq` and `lt`
* Refactor for `lt` and `eq` where the only difference is the jump-command

Next, we'll continue working on adding more commands to `StackTest.vm`.

### CODE REVIEW

**Review by tomstuart — APPROVED** [review]
Excellent! Great job on the refactoring — it’s really helpful to see the separate steps of “make different parts identical” followed by “eliminate duplication”.

This is totally mergeable; the review comments are the usual optional chit-chat.

File: lib/code_writer.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class CodeWriter
   def initialize(out)
-    @out = out 
+    @out = out
+    @label_counter = 0
```
[the remark this answers is not in the document]
Comment (wildmaples) [reply lib/code_writer.rb:4]: Neat! Using `Enumerator` would be useful if we wanted to do additional work when counting up. Also a fun reminder that `Enumerator` != `Enumerable`.

File: lib/code_writer.rb [hunk 2]
```diff
@@ -25,38 +25,16 @@ def write_arithmetic(command)
         A=A-1
         M=M#{operation}D
       EOF
-    elsif command == "eq"
+    elsif command == "eq" or command == "lt"
+      jump = command == "eq" ? "JEQ" : "JLT"
```
[the remark this answers is not in the document]
Comment (wildmaples) [reply lib/code_writer.rb]: [`83826d3` (#5)](https://github.com/wildmaples/jack-virtual-machine/pull/5/commits/83826d3b48e459329259483ee00365d77405b21c) ✅ ❓

[the remark this answers is not in the document]
Comment (tomstuart) [reply lib/code_writer.rb]: Nice! 🎉

**Pull request merged**